streams/src/test/java/org/apache/kafka/common/metrics/SensorAccessor.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.metrics;
+
```

Comment:
This accessor enables the unit test to access package-private member `parents()` of class `Sensor` without the need to modify `Sensor`'s public API. (kudos to @vvcephei for the idea)

Reply:
o.a.k.common.metrics is considered a semi-public package, do we have to make this package?

Reply:
Does semi-public package mean, we do not need a KIP to change it? My first push to this PR just made `parents()` public instead of package-private. Afterwards I realized that Sensor is a public API.

Reply:
Actually I think I'm wrong: checking https://kafka.apache.org/24/javadoc/index.html here the o.a.k.common.metrics is not listed, only `Metric` and `MetricName` under o.a.k.common are listed in javadocs. So this should be fine.

Reply:
But wouldn't that mean, I could simply make `parents()` public without the need of the accessor class?

Reply:
Yeah I think that's right: `Sensor` is not a public API either.